README.md

```diff
@@ -28,9 +28,13 @@ Example content:
 
 ```json
 {
-    "host": "localhost",
-    "port": 8086,
-    "secretkey": "mysecretkey"
+  "host": "localhost",
+  "port": 8086,
+  "secretkey": "mysecretkey",
+  "ssl": {
+    "cert_path": "/path/to/cert.pem",
+    "key_path": "/path/to/key.pem"
+  }
 }
 ```
 
```

Comment:
* Technically then "all fields are optional" are false, since if cert_path is specified then key_path must be specified, right?
* Maybe edit the part above to change `ws://` to "`ws://` (or `wss://` if `ssl` key is nonempty)"

Reply:
Actually all fields remain optional because of the following:

We have in the [config.py file](https://github.com/CosmicDNA/plover_websocket_server/blob/8471b075a45b8ae43e0198fbbcbdef4fe323d0f5/plover_engine_server/config.py#L40):

```python
self.ssl = data.get('ssl', {})
```

so if there is no `ssl` in the config file it gets an empty `dict` `{}`.

Then we have in the [websocket/server.py file](https://github.com/CosmicDNA/plover_websocket_server/blob/8471b075a45b8ae43e0198fbbcbdef4fe323d0f5/plover_engine_server/websocket/server.py#L66)

```python
if self._ssl:
    # Load your SSL certificate and private key
    ssl_context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
    ssl_context.load_cert_chain(self._ssl.get('cert_path'), self._ssl.get('key_path'))
else:
    ssl_context = None
```

so if the `_ssl` `dict` is empty as depicted by `{}`, it will set the `ssl_context` to `None` as is the default behaviour when the parameter `ssl_context` is not specified on the [`web.TCPSite` constructor](https://github.com/CosmicDNA/plover_websocket_server/blob/8471b075a45b8ae43e0198fbbcbdef4fe323d0f5/plover_engine_server/websocket/server.py#L73).

Reply:
To answer your second concern, I have added the secret key to a [middleware in websocket/server.py](https://github.com/CosmicDNA/plover_websocket_server/blob/b7f5a38ab1daa04a3cfac1eab23a4cabe6c3c1e0/plover_engine_server/websocket/server.py#L40). This way all the requests are tested against the secretkey. And additionally I have added a second middleware which injects the `ssl` context to the request in such a way to allow the [protocol handle](https://github.com/CosmicDNA/plover_websocket_server/blob/4bfc6d3f3946d2149675a684fce272602b7f42a9/plover_engine_server/websocket/views.py#L16) to reply with the correct protocol in place.